src/core/statistics_chain.cpp

```diff
@@ -106,26 +106,25 @@ void calc_rg(PartCfg &partCfg, double **_rg) {
     r_G2 += tmp;
     r_G4 += tmp * tmp;
   }
-  tmp = (double)chain_n_chains;
+  tmp = static_cast<double>(chain_n_chains);
   rg[0] = r_G / tmp;
   rg[2] = r_G2 / tmp;
   rg[1] = sqrt(rg[2] - rg[0] * rg[0]);
   rg[3] = sqrt(r_G4 / tmp - rg[2] * rg[2]);
+  return rg;
 }
 
-void calc_rh(PartCfg &partCfg, double **_rh) {
-  int i, j, p;
-  double dx, dy, dz, r_H = 0.0, r_H2 = 0.0, *rh = nullptr, ri = 0.0, prefac,
-                     tmp;
-  *_rh = rh = Utils::realloc(rh, 2 * sizeof(double));
+std::vector<double> calc_rh(PartCfg &partCfg) {
```

Comment:
Shouldn't be on the heap, the size is known.

Reply:
yep